### PR TITLE
Add docstring for `BlockChain.ProposeBlock()`

### PR DESCRIPTION
**Pull request by limebell**

### CODE REVIEW

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 1]
```diff
@@ -258,6 +287,29 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
...
+        /// <summary>
+        /// Proposes a next <see cref="Block{T}"/> using staged <see cref="Transaction{T}"/>s.
+        /// </summary>
+        /// <param name="proposer">
+        /// The proposer's <see cref="PublicKey"/> that proposes the block.</param>
+        /// <param name="timestamp">The <see cref="DateTimeOffset"/> when proposing started.</param>
+        /// <param name="maxBlockBytes">The maximum number of bytes a block can have.
+        /// See also <see cref="IBlockPolicy{T}.GetMaxBlockBytes(long)"/>.</param>
+        /// <param name="maxTransactions">The maximum number of transactions that a block can
+        /// accept.  See also <see cref="IBlockPolicy{T}.GetMaxTransactionsPerBlock(long)"/>.
```
Comment (colibrishin) [inline Libplanet/Blockchain/BlockChain.MineBlock.cs:299]:
```suggestion
        /// accept. See also <see cref="IBlockPolicy{T}.GetMaxTransactionsPerBlock(long)"/>.
```

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 2]
```diff
@@ -258,6 +287,29 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
...
+        /// </param>
+        /// <param name="maxTransactionsPerSigner">The maximum number of transactions
+        /// that a block can accept per signer.  See also
```
Comment (colibrishin) [inline Libplanet/Blockchain/BlockChain.MineBlock.cs:302]:
```suggestion
        /// that a block can accept per signer. See also
```

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 3]
```diff
@@ -258,6 +287,29 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
...
+        /// <see cref="IBlockPolicy{T}.GetMaxTransactionsPerSignerPerBlock(long)"/>.</param>
+        /// <param name="txPriority">An optional comparer for give certain transactions to
+        /// priority to belong to the block.  No certain priority by default.</param>
```
Comment (colibrishin) [inline Libplanet/Blockchain/BlockChain.MineBlock.cs:305]:
```suggestion
        /// <param name="txPriority">An optional comparer for prioritizing certain transactions
        /// that are belong to the block. No certain priority by default.</param>
```

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 4]
```diff
@@ -258,6 +287,29 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
...
+        /// <param name="lastCommit"><see cref="BlockCommit"/> of previous <see cref="Block{T}"/>.
```
Comment (colibrishin) [inline Libplanet/Blockchain/BlockChain.MineBlock.cs:306]:
```suggestion
        /// <param name="lastCommit">A <see cref="BlockCommit"/> of previous <see cref="Block{T}"/>.
```

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 5]
```diff
@@ -258,6 +287,29 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
...
+        /// </param>
+        /// <param name="cancellationToken">A cancellation token used to propagate notification
```
Comment (colibrishin) [inline Libplanet/Blockchain/BlockChain.MineBlock.cs:308]:
```suggestion
        /// <param name="cancellationToken">A cancellation token used to propagates notification
```

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 6]
```diff
@@ -258,6 +287,29 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
                 cancellationToken: cancellationToken ?? default(CancellationToken));
 #pragma warning restore SA1118
 
+        /// <summary>
+        /// Proposes a next <see cref="Block{T}"/> using staged <see cref="Transaction{T}"/>s.
```
Comment (colibrishin) [inline Libplanet/Blockchain/BlockChain.MineBlock.cs:291]:
```suggestion
        /// Proposes a next <see cref="Block{T}"/> with staged <see cref="Transaction{T}"/>s.
```

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 7]
```diff
@@ -233,10 +233,39 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
             return block;
         }
 
+        /// <summary>
+        /// <para>
+        /// Proposes a next <see cref="Block{T}"/> using staged <see cref="Transaction{T}"/>s.
```
Comment (colibrishin) [inline Libplanet/Blockchain/BlockChain.MineBlock.cs:238]:
```suggestion
        /// Proposes a next <see cref="Block{T}"/> with staged <see cref="Transaction{T}"/>s.
```

**Review by colibrishin — CHANGES_REQUESTED** [review]

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 8]
```diff
@@ -233,10 +233,39 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
...
+        /// <param name="maxBlockBytes">The maximum number of bytes a block can have.
+        /// See also <see cref="IBlockPolicy{T}.GetMaxBlockBytes(long)"/>.</param>
+        /// <param name="maxTransactions">The maximum number of transactions that a block can
+        /// accept.  See also <see cref="IBlockPolicy{T}.GetMaxTransactionsPerBlock(long)"/>.
+        /// </param>
+        /// <param name="maxTransactionsPerSigner">The maximum number of transactions
+        /// that a block can accept per signer.  See also
+        /// <see cref="IBlockPolicy{T}.GetMaxTransactionsPerSignerPerBlock(long)"/>.</param>
+        /// <param name="txPriority">An optional comparer for give certain transactions to
+        /// priority to belong to the block.  No certain priority by default.</param>
```
[the remark this answers is not in the document]
Comment (limebell) [reply Libplanet/Blockchain/BlockChain.MineBlock.cs:258]: It's copy of `BlockChain.Mine()` docstring. Do you think it should be changed to?

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 9]
```diff
@@ -233,10 +233,39 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
...
+        /// block is produced with current timestamp and appended immediately to the chain.
+        /// </para>
+        /// </summary>
+        /// <param name="proposer">
+        /// The proposer's <see cref="PublicKey"/> that proposes the block.</param>
+        /// <param name="timestamp">The <see cref="DateTimeOffset"/> when proposing started.</param>
+        /// <param name="maxBlockBytes">The maximum number of bytes a block can have.
+        /// See also <see cref="IBlockPolicy{T}.GetMaxBlockBytes(long)"/>.</param>
+        /// <param name="maxTransactions">The maximum number of transactions that a block can
+        /// accept.  See also <see cref="IBlockPolicy{T}.GetMaxTransactionsPerBlock(long)"/>.
```
[the remark this answers is not in the document]
Comment (limebell) [reply Libplanet/Blockchain/BlockChain.MineBlock.cs:252]: This is intended, it's our convention for writing docstring and separate paragraphs.
See also:
https://github.com/planetarium/libplanet/blob/main/Libplanet/Blockchain/BlockChain.MineBlock.cs#L83

File: Libplanet/Blockchain/BlockChain.MineBlock.cs [hunk 10]
```diff
@@ -233,10 +233,39 @@ void WatchTip(object target, (Block<T> OldTip, Block<T> NewTip) tip)
...
+        /// </param>
+        /// <param name="maxTransactionsPerSigner">The maximum number of transactions
+        /// that a block can accept per signer.  See also
+        /// <see cref="IBlockPolicy{T}.GetMaxTransactionsPerSignerPerBlock(long)"/>.</param>
+        /// <param name="txPriority">An optional comparer for give certain transactions to
+        /// priority to belong to the block.  No certain priority by default.</param>
+        /// <param name="lastCommit"><see cref="BlockCommit"/> of previous <see cref="Block{T}"/>.
+        /// </param>
+        /// <param name="cancellationToken">A cancellation token used to propagate notification
+        /// that this operation should be canceled.</param>
```
[the remark this answers is not in the document]
Comment (limebell) [reply Libplanet/Blockchain/BlockChain.MineBlock.cs:262]: I'm not really a master of English grammar, many use case in our docstrings for the parameter `cancellationToken` is this.

[the remark this answers is not in the document]
Comment (colibrishin) [reply Libplanet/Blockchain/BlockChain.MineBlock.cs:262]: Oh, former grammer was right.

**Review by colibrishin — APPROVED** [review]

**Pull request merged**